### DISCUSSION TITLE
right-drawer props(): Quasar Issue?

### DISCUSSION THREAD

**Discussion by kwmartin** [zauberzeug/nicegui#2862] [Q&A]
### Description

If this is the wrong venue, please advise. I tried Discord but invite didn't work, and I didn't take the time to figure out why; I will if that's the correct venue
Anyways:
```py
with ui.right_drawer(fixed=True).style('background-color: #ebf1fa').props('bordered overlay show-if-above value=False') as right_drawer:
```
with respect to show-if-above as shown right_drawer is open when routing or updating the page, but when setting a value using such as:
```py
with ui.right_drawer(fixed=True).style('background-color: #ebf1fa').props('bordered overlay show-if-above=None value=False') as right_drawer:
```
show-if-above='None' is set (note the quotes). Similar for True and False, that is the values are strings not <class 'bool'> and these seem to be interpreted as truthy? Also, as best I can understand the inline documentation and looking at the code, and experimenting, my best guess to add props is something like:
```py
right_drawer.props(add='show-if-above=None value=False')
```
and examining the attributes of right_drawer, we see:
```py
(Pdb) right_drawer.props(add='show-if-above=None value=False')
<nicegui.page_layout.RightDrawer object at 0x77cc58c71ed0>
(Pdb) pi right_drawer.props()
right_drawer.props(). ___visible = True
right_drawer.props(). ignores_events_when_hidden = True
right_drawer.props(). client = <nicegui.client.Client object at 0x77cc58c70af0>
right_drawer.props(). id = 16
right_drawer.props(). tag = q-drawer
right_drawer.props(). _classes = ['nicegui-drawer']
right_drawer.props(). _style = {'background-color': '#ebf1fa'}
right_drawer.props(). _props = {'show-if-above': 'None', 'side': 'right', 'bordered': True, 'elevated': False, 'overlay': True, 'value': 'False'}
right_drawer.props(). _event_listeners = {}
right_drawer.props(). _text = None
right_drawer.props(). slots = {'default': <nicegui.slot.Slot object at 0x77cc58c71ea0>}
right_drawer.props(). default_slot = <nicegui.slot.Slot object at 0x77cc58c71ea0>
right_drawer.props(). _deleted = False
right_drawer.props(). parent_slot = <nicegui.slot.Slot object at 0x77cc58c70be0>
right_drawer.props(). tailwind = <nicegui.tailwind.Tailwind object at 0x77cc58c71f00>
(Pdb) 
```
and the props 'show-if-above' and 'value' are respectively: 'None' and 'False', again strings. I find this perplexing and can't understand the reasoning. Are there any explanations or examples on adding/removing Quasar properties? Also, an example of adding an additional Quasar structure would be very useful. Finally (this is a Quasar issue I think), does anyone know how to use the 'value' property? and what choices are reasonable? I've been unsuccessful in figuring this out by reading the Quasar docs. Thanks.

**Comment by falkoschindler** [comment]
Hi @kwmartin,

Boolean props like "bordered", "overlay" or "show-if-above" are set to `True` simply by adding them like this:
```py
with ui.right_drawer().props('bordered overlay show-if-above'):
    ui.label('Right drawer')
```
To set the `value` (i.e. the opening state or the drawer), you can simply pass it to the initializer:
```py
with ui.right_drawer(value=False).props('bordered overlay show-if-above'):
    ui.label('Right drawer')
```
But note that "show-if-above" basically overwrites `value`, which is why the drawer is still visible on large screens:
> **show-if-above:** Forces drawer to be shown on screen on initial render if the layout width is above breakpoint, regardless of v-model; This is the default behavior when SSR is taken over by client on initial render
https://quasar.dev/layout/drawer

**Reply by kwmartin** [reply]
From the Quasar docs:
> Forces drawer to be shown on screen on initial render if the layout width is above breakpoint, regardless of v-model; This is the default behavior when SSR is taken over by client on initial render

which seems to tell me the default is True. I can't see how one sets show-if-above == False. Also, under documentation ui.right_drawer, value is not included as one of the parameters? It also is not listed as a parameter in Quasar documentation. I have also looked in the code, and I can't see where to look to understand what "value" does and where and how it should be set. Still trying.

**Reply by falkoschindler** [reply]
You're right, the documentation for `ui.right_drawer` could be improved. Currently there is only a page on [page layout](https://nicegui.io/documentation/page_layout) that doesn't list the initializer arguments of each layout element.

But there's a description in the docstring of `ui.right_drawer.__init__` which should be shown as a tooltip in your IDE:
<img width="784" alt="Screenshot 2024-04-11 at 07 54 14" src="https://github.com/zauberzeug/nicegui/assets/5767091/6728e78b-113c-4b5a-8420-b8ec2a2de800">
<img width="777" alt="Screenshot 2024-04-11 at 07 54 20" src="https://github.com/zauberzeug/nicegui/assets/5767091/3f95ccac-9feb-4883-b6bf-a7b99e082d5b">

Regarding the statement "This is the default behavior when SSR is taken over by client on initial render": I think this only applies to SSR (server-side rendering), which is not the case for NiceGUI apps. Quasar's documentation could be a bit more explicit about default values of those props. But these boolean flags are usually false by default. In HTML you set them to true by simply mentioning them, like "round" and "dense" in the following example:
```html
<q-btn flat @click="drawer = !drawer" round dense icon="menu" />
```
And you set them to false by _not_ mentioning them. This is what we mimicked with NiceGUI.

Of course we don't expect our users to know Vue and Quasar. I'm just giving some background information here.
